### PR TITLE
[MI-2340] Fix issues given in feedback and fix height issue in commen…

### PR DESCRIPTION
**Pull request by ayusht2810**
…t modal

1. Add a title in shared records to redirect to ServiceNow URL
2. Redirecting shared records links to ServiceNow
3. Update links in the modal to redirect to ServiceNow
4. Fix the issue of the button not opening the modal for notification posts
6. Add a feature to select dropdown values using the keyboard keys
7. Pre-fill the channel field on opening the subscription modal
8. Fix height issue in the comment modal

### CODE REVIEW

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/api.go [hunk 1]
```diff
@@ -439,6 +439,13 @@ func (p *Plugin) getRecordFromServiceNow(w http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	record.RecordType = recordType
+	if err := record.HandleNestedFields(p.getConfiguration().ServiceNowBaseURL); err != nil {
+		p.API.LogError("Invalid body", "Error", err.Error())
+		p.handleAPIError(w, &serializer.APIErrorResponse{StatusCode: http.StatusBadRequest, Message: fmt.Sprintf("Invalid body. Error: %s", err.Error())})
```
Comment (manojmalik20) [inline server/plugin/api.go:445]: How can we send `BadRequest`? This is the response we got from ServiceNow. 
```suggestion
		p.API.LogError("Error in handling the nested fields", "Error", err.Error())
		p.handleAPIError(w, &serializer.APIErrorResponse{StatusCode: http.StatusInternalServerError, Message: fmt.Sprintf("Error in handling the nested fields. Error: %s", err.Error())})
```

File: server/plugin/api.go [hunk 2]
```diff
@@ -479,12 +486,6 @@ func (p *Plugin) shareRecordInChannel(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := record.HandleNestedFields(); err != nil {
```
Comment (manojmalik20) [inline server/plugin/api.go:482]: Why removed this?

File: server/serializer/servicenow_record.go [hunk 3]
```diff
@@ -156,27 +157,27 @@ func (sr *ServiceNowRecord) CreateSharingPost(channelID, botID, serviceNowURL, p
...
 		if err != nil {
 			return fmt.Errorf("%w : kb_knowledge_base", err)
 		}
-		sr.Category, err = GetNestedFieldValue(sr.Category)
+		sr.Category, err = GetNestedFieldValue(sr.Category, constants.FieldCategory, serviceNowURL)
 		if err != nil {
 			return fmt.Errorf("%w : kb_category", err)
 		}
-		sr.Author, err = GetNestedFieldValue(sr.Author)
+		sr.Author, err = GetNestedFieldValue(sr.Author, constants.FieldAssignedTo, serviceNowURL)
```
Comment (manojmalik20) [inline server/serializer/servicenow_record.go:171]: You are passing `constants.FieldAssignedTo` and assigning value in `Author`. Is this right?

File: webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx [hunk 4]
```diff
@@ -97,6 +98,12 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
     };
 
     useEffect(() => {
+        if (open) {
+            if (currentChannelId) {
```
Comment (manojmalik20) [inline webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx:102]:
```suggestion
        if (open && currentChannelId) {
```

File: webapp/src/utils/index.tsx [hunk 5]
```diff
@@ -62,6 +62,27 @@ export const onPressingEnterKey = (event: React.KeyboardEvent<HTMLSpanElement> |
...
+    if (key === KnowledgeRecordDataLabelConfigKey.KNOWLEDGE_BASE ||
+        key === KnowledgeRecordDataLabelConfigKey.AUTHOR ||
+        key === KnowledgeRecordDataLabelConfigKey.CATEGORY ||
+        key === RecordDataLabelConfigKey.ASSIGNED_TO ||
+        key === RecordDataLabelConfigKey.ASSIGNMENT_GROUP
+    ) {
+        return true;
+    }
+
+    return false;
```
Comment (manojmalik20) [inline webapp/src/utils/index.tsx:83]:
```suggestion
    return key === KnowledgeRecordDataLabelConfigKey.KNOWLEDGE_BASE ||
        key === KnowledgeRecordDataLabelConfigKey.AUTHOR ||
        key === KnowledgeRecordDataLabelConfigKey.CATEGORY ||
        key === RecordDataLabelConfigKey.ASSIGNED_TO ||
        key === RecordDataLabelConfigKey.ASSIGNMENT_GROUP;
```

File: webapp/src/utils/index.tsx [hunk 6]
```diff
@@ -62,6 +62,27 @@ export const onPressingEnterKey = (event: React.KeyboardEvent<HTMLSpanElement> |
...
 
+export const getLinkData = (value: string): LinkData => {
+    const data = value.split(']');
+    return ({
+        display_value: data[0].slice(1),
+        link: data[1].slice(1, -1),
+    });
+};
+
+export const validateKeys = (key: string) => {
```
Comment (manojmalik20) [inline webapp/src/utils/index.tsx:73]:
```suggestion
export const validateKeysContainingLink = (key: string) => {
```

File: server/serializer/servicenow_record.go (hunk 3 above)
Comment (ayusht2810) [reply server/serializer/servicenow_record.go:171]: yes both use the same constant

File: server/plugin/api.go (hunk 2 above)
Comment (ayusht2810) [reply server/plugin/api.go:482]: we are directly sending formatted data in the request so we don't need this

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**